packages/core-components/src/layout/SignInPage/guestProvider.tsx

```diff
@@ -20,39 +20,55 @@ import Button from '@material-ui/core/Button';
 import { InfoCard } from '../InfoCard/InfoCard';
 import { GridItem } from './styles';
 import { ProviderComponent, ProviderLoader, SignInProvider } from './types';
-import { GuestUserIdentity } from './GuestUserIdentity';
-
-const Component: ProviderComponent = ({ onSignInStarted, onSignInSuccess }) => (
-  <GridItem>
-    <InfoCard
-      title="Guest"
-      variant="fullHeight"
-      actions={
-        <Button
-          color="primary"
-          variant="outlined"
-          onClick={() => {
-            onSignInStarted();
-            onSignInSuccess(new GuestUserIdentity());
-          }}
-        >
-          Enter
-        </Button>
-      }
-    >
-      <Typography variant="body1">
-        Enter as a Guest User.
-        <br />
-        You will not have a verified identity,
-        <br />
-        meaning some features might be unavailable.
-      </Typography>
-    </InfoCard>
-  </GridItem>
-);
-
-const loader: ProviderLoader = async () => {
-  return new GuestUserIdentity();
+import { ProxiedSignInIdentity } from '../ProxiedSignInPage/ProxiedSignInIdentity';
+import { discoveryApiRef, useApi } from '@backstage/core-plugin-api';
+
+const Component: ProviderComponent = ({ onSignInStarted, onSignInSuccess }) => {
+  const discoveryApi = useApi(discoveryApiRef);
+  return (
+    <GridItem>
+      <InfoCard
+        title="Guest"
+        variant="fullHeight"
+        actions={
+          <Button
+            color="primary"
+            variant="outlined"
+            onClick={() => {
+              onSignInStarted();
+              onSignInSuccess(
+                new ProxiedSignInIdentity({
+                  provider: 'guest',
+                  discoveryApi,
+                }),
```

Comment:
So, if we put this code here, everybody who's on the old backend system will no longer be able to use guest auth.

Could you make it so that it first tries the proxy auth, and if it throws / doesn't return an identity (see the comment in `loader`), it opens a popup (even a silly `confirm()`) asking the user "The guest auth provider is not installed, do you want to continue with a classic token-less guest identity?"

If you do so, you may want to like flag in local storage or whatnot, what the user choice was, because the loader will need to know next time around which one of the identities to try. Essentially if the user responded yes, it'll just return GuestUserIdentity like before.

Reply:
Yup, good call. Given that _most_ installations probably have this set up in one form or another, I think backporting this module to the old backend wouldn't fix this issue as I was originally thinking. In that vein, I've updated this to now check if you have the new backend module installed, if yes, we use the proxied sign in. If not, we show a confirm and then store the positive result in local storage. If they choose the negative option, "cancel" here, we just trigger a signin failure, as it's an impossible to use Backstage instance with no identity.

Then on load, we check if a) they have the backend module set up, if so we use proxy auth, b) if they have the guest override set _and_ the new backend module, if so we try to get them to choose one and  will not remember their choice (we can change this, I'm not 100% why people would want to continue using the old frontend-only tokens), c) if they have the guest override set and no valid identity from the backend module (either not installed or not configured correctly), we use the old way of authenticating. 

This does mean we're now making a single API request per load in the guest account, but I think the experience for folks wanting to transfer over to the new module will be better this way. They won't have to dig around in their local storage to remove items.